pkg/traces/servicegraphprocessor/processor.go

```diff
@@ -158,21 +191,38 @@ func (p *processor) registerMetrics() error {
 		}
 	}
 
-	// Collect unpaired spans when evicting items from the store during
-	// periodic cleanup
-	p.store.OnEvicted(func(s string, i interface{}) {
-		e := i.(edgeRequest)
-		if !e.complete() {
-			p.serviceGraphUnpairedSpansTotal.WithLabelValues(e.clientService, e.serverService).Inc()
+	go func() {
```

Comment:
what throughput have you tested this at? at some point your channel is going to back up and you are going to start refusing traces.

should we have a configurable number of workers?

Reply:
I've only tested it at ~6500 spans/sec, not very high. I've the logic to spawn multiple workers.